code/API_definitions/qod-api.yaml

```diff
@@ -290,6 +290,66 @@ paths:
         - threeLegged:
             - "qod-sessions-delete"
 
+  /sessions/{sessionId}/extend:
+    post:
+      tags:
+        - Change session duration
+      summary: "Extend the duration of an active session"
+      description: |
+        Extend the duration of an active QoS session.
+        If this operation is executed successfully, the new duration of the target session will be the original duration plus extended duration.
+      operationId: extendQosSessionDuration
+      parameters:
+        - name: sessionId
+          in: path
+          description: Session ID that was obtained from the createSession operation
+          required: true
+          schema:
+            $ref: "#/components/schemas/SessionId"
+      requestBody:
+        description: Parameters to extend the duration of an active session
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/ExtendSessionDuration"
+        required: true
+      responses:
+        "200":
+          description: Contains information about active session
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/SessionInfo"
+        "400":
+          description: Invalid input for extendQosSessionDuration operation
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorInfo"
+              examples:
+                Generic400:
+                  summary: Some parameter combinations or parameter values provided are not schema compliant
+                  value:
+                    status: 400
+                    code: INVALID_ARGUMENT
+                    message: "Schema validation failed at  ..."
+                InactiveSession:
+                  summary: The target session is inactive
+                  value:
+                    status: 400
+                    code: INVALID_ARGUMENT
+                    message: "The target session is inactive"
```

Comment:
It would be useful to include an example of an error when the required extension is over the maximum session duration. This is something that we should also make clear in the description, if remaining + extension <= max session duration.

Reply:
It is added.